EIPS/eip-2386.md

```diff
@@ -0,0 +1,201 @@
+---
+eip: 2386
+title: Ethereum 2 Hierarchical Deterministic Walletstore
+author: Jim McDonald <Jim@mcdee.net>
+discussions-to: https://ethereum-magicians.org/t/eip-2386-walletstore/3792
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-11-21
+requires: 2334, 2335
+---
+
+## Simple Summary
+
+A JSON format for the storage and retrieval of Ethereum 2 hierarchical deterministic (HD) wallet definitions.
+
+## Abstract
+
+Ethereum has the concept of keystores: pieces of data that define a key (see [EIP-2335](https://eips.ethereum.org/EIPS/eip-2335) for details).  This adds the concept of walletstores: stores that define wallets and how keys in said wallets are created.
+
+## Motivation
+
+Hierarchical deterministic wallets create keys from a _seed_ and a _path_.  The seed needs to be accessible to create new keys, however it should also be protected to the same extent as private keys to stop it from becoming an easy attack vector.  The path, or at least the variable part of it, needs to be stored to ensure that keys are not duplicated.  Providing a standard method to do this can promote interoperability between wallets and similar software.
+
+Given that a wallet has an amount of data and metadata that is useful when accessing existing keys and creating new keys, standardizing this information and how it is stored allows it to be portable between different wallet providers with minimal effort.
+
+## Specification
+
+The elements of a hierarchical deterministic walletstore are as follows:
+
+### UUID
+
+The `uuid` provided in the walletstore is a randomly-generated type 4 UUID as specified by [RFC 4122](https://tools.ietf.org/html/rfc4122). It is intended to be used as a 128-bit proxy for referring to a particular wallet, used to uniquely identify wallets.
+
+This element MUST be present.  It MUST be a string following the syntactic structure as laid out in [section 3 of RFC 4122](https://tools.ietf.org/html/rfc4122#section-3).
```

Comment:
Consider `**MUST**` instead of `MUST` so they stand out a bit more (this is also the dominant style in this repository).